provider/vsphere/environ_broker.go

```diff
@@ -289,9 +290,13 @@ func (env *sessionEnviron) AllInstances(ctx context.ProviderCallContext) ([]inst
 
 	// Turn mo.VirtualMachine values into *environInstance values,
 	// whether or not we got an error.
-	results := make([]instances.Instance, len(vms))
-	for i, vm := range vms {
-		results[i] = newInstance(vm, env.environ)
+	results := make([]instances.Instance, 0, len(vms))
```

Comment:
Overkill for a small list I think.
Can just be:
`var results []instances.Instance`

There should be a unit test update also

Reply:
Cleaned up, thanks.